File: src/main/java/com/slowgenius/toolkit/base/VisibleConfig.java
package com.slowgenius.toolkit.base;

import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.ex.ActionUtil;
import com.intellij.openapi.components.PersistentStateComponent;
import com.intellij.openapi.components.State;
import com.intellij.openapi.components.Storage;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.project.ProjectManager;
import com.slowgenius.toolkit.autoCreate.CreateClasses;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.HashMap;
import java.util.Map;

/**
 * 插件是否可见配置
 *
 * @author hty
 * date: 2022/11/9
 */
@State(name = "visibleConfig", storages = {@Storage(value = "visibleConfig.xml")})
public class VisibleConfig implements PersistentStateComponent<VisibleConfig> {

    public static VisibleConfig INSTANCE = new VisibleConfig();
    private Map<String, Boolean> properties = new HashMap<>();


    public Map<String, Boolean> getProperties() {
        return properties;
    }

    public void setProperties(Map<String, Boolean> properties) {
        this.properties = properties;
    }

    @Override
    public @Nullable VisibleConfig getState() {
        if (this.properties.isEmpty()) {
            CreateClasses action = (CreateClasses) ActionUtil.getAction("CreateClasses");
            properties.put(action.getActionKey(), true);
        }
        return this;
    }

    @Override
    public void loadState(@NotNull VisibleConfig visibleConfig) {
        if (visibleConfig.properties == null || visibleConfig.properties.isEmpty()) {
            visibleConfig.properties = new HashMap<>();
            CreateClasses action = (CreateClasses) ActionUtil.getAction("CreateClasses");
            properties.put(action.getActionKey(), true);
        }
        this.properties = visibleConfig.properties;
    }

    public static VisibleConfig getInstance(Project project) {
        return project.getService(VisibleConfig.class);
    }

    public static VisibleConfig getInstance() {
        return ProjectManager.getInstance().getDefaultProject().getService(VisibleConfig.class);
    }
}

File: src/main/java/com/slowgenius/toolkit/base/VisibleConfig.kt
package com.slowgenius.toolkit.base

import com.intellij.openapi.actionSystem.ActionManager
import com.intellij.openapi.application.ApplicationManager
import com.intellij.openapi.components.PersistentStateComponent
import com.intellij.openapi.components.State
import com.intellij.openapi.components.Storage
import com.slowgenius.toolkit.autoCreate.CreateClasses
import java.util.*
import kotlin.collections.HashMap

/**
 * 插件是否可见配置
 *
 * @author hty
 * date: 2022/11/9
 */
@State(name = "visibleConfig", storages = [Storage(value = "visibleConfig.xml")])
class VisibleConfig : PersistentStateComponent<VisibleConfig?> {
    var properties: MutableMap<String, Boolean> = HashMap()

    override fun getState(): VisibleConfig {
        return this
    }

    override fun loadState(visibleConfig: VisibleConfig) {
        properties = visibleConfig.properties
    }

    private fun init() {
        properties = HashMap()
        val action = (Optional.ofNullable(ActionManager.getInstance()).map { item-> item.getAction("intelli create") }.orElse(null) as CreateClasses)
        properties[action.getActionKey()] = true
    }

    companion object {
        @JvmStatic
           fun getInstance(): VisibleConfig {
                val visibleConfig = ApplicationManager.getApplication().getService(VisibleConfig::class.java)
                if (visibleConfig.properties.isEmpty()) {
                    visibleConfig.init()
                }
                return visibleConfig
            }
    }
}